### ISSUE TITLE
Has hidden dependency on rails-ujs

### ISSUE THREAD

**Issue by bf4** [alvesjtiago/fastentry#7]
We just tried to deploy this on Rails 5.0 and got a failure on precompile

>        Sprockets::FileNotFound: couldn't find file 'rails-ujs' with type 'application/javascript'

per https://github.com/rails/rails-ujs/tree/master it was merged into Rails 5.1 and on Rails 5.0 we need to install it by, for example, `gem 'rails-ujs'`

```
       I, [2019-08-01T23:04:26.805920 #368]  INFO -- : Writing /tmp/build_d8e573ed04088d976e972a9313d20c49/public/assets/fastentry/logo-ad1f85a9141125a4db1515c97e98bb768bd1fc4e5d70f5061ece2d6a4d895b18.png
       rake aborted!
       Sprockets::FileNotFound: couldn't find file 'rails-ujs' with type 'application/javascript'
       Checked in these paths: 
         /tmp/build_d8e573ed04088d976e972a9313d20c49/app/assets/images
         /tmp/build_d8e573ed04088d976e972a9313d20c49/app/assets/javascripts
         /tmp/build_d8e573ed04088d976e972a9313d20c49/app/assets/stylesheets
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/assets/javascripts
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/assets/stylesheets
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/logster-2.3.0/vendor/assets/javascripts
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/fastentry-0.1.15/app/assets/config
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/fastentry-0.1.15/app/assets/images
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/fastentry-0.1.15/app/assets/javascripts
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/fastentry-0.1.15/app/assets/stylesheets
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/bundler/gems/apitome-75b1d243186b/app/assets/javascripts
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/bundler/gems/apitome-75b1d243186b/app/assets/stylesheets
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/bundler/gems/apitome-75b1d243186b/vendor/assets/javascripts
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/bundler/gems/apitome-75b1d243186b/vendor/assets/stylesheets
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/pghero-2.2.1/app/assets/images
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/pghero-2.2.1/app/assets/javascripts
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/pghero-2.2.1/app/assets/stylesheets
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/bundler/gems/apitome-75b1d243186b/assets/stylesheets
         /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/bundler/gems/apitome-75b1d243186b/assets/javascripts
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/fastentry-0.1.15/app/assets/javascripts/fastentry/application.js:13
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/resolve.rb:65:in `resolve!'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/directive_processor.rb:399:in `resolve'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/directive_processor.rb:207:in `process_require_directive'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/directive_processor.rb:180:in `block in process_directives'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/directive_processor.rb:178:in `each'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/directive_processor.rb:178:in `process_directives'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/directive_processor.rb:83:in `_call'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/directive_processor.rb:68:in `call'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:75:in `call_processor'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:57:in `block in call_processors'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:56:in `reverse_each'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:56:in `call_processors'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/loader.rb:134:in `load_from_unloaded'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/loader.rb:60:in `block in load'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/loader.rb:317:in `fetch_asset_from_dependency_cache'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/loader.rb:44:in `load'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/cached_environment.rb:20:in `block in initialize'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/cached_environment.rb:47:in `load'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/bundle.rb:23:in `block in call'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/utils.rb:200:in `dfs'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/bundle.rb:24:in `call'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:75:in `call_processor'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:57:in `block in call_processors'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:56:in `reverse_each'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:56:in `call_processors'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/loader.rb:134:in `load_from_unloaded'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/loader.rb:60:in `block in load'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/loader.rb:317:in `fetch_asset_from_dependency_cache'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/loader.rb:44:in `load'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/cached_environment.rb:20:in `block in initialize'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/cached_environment.rb:47:in `load'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/base.rb:66:in `find_asset'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/base.rb:73:in `find_all_linked_assets'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/manifest.rb:142:in `block in find'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/legacy.rb:114:in `block (2 levels) in logical_paths'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:228:in `block in stat_tree'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:212:in `block in stat_directory'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:209:in `each'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:209:in `stat_directory'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:227:in `stat_tree'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:231:in `block in stat_tree'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:212:in `block in stat_directory'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:209:in `each'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:209:in `stat_directory'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/path_utils.rb:227:in `stat_tree'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/legacy.rb:105:in `each'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/legacy.rb:105:in `block in logical_paths'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/legacy.rb:104:in `each'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/legacy.rb:104:in `logical_paths'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/manifest.rb:140:in `find'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/sprockets/manifest.rb:186:in `compile'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-rails-3.2.1/lib/sprockets/rails/task.rb:68:in `block (3 levels) in define'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-3.7.2/lib/rake/sprocketstask.rb:147:in `with_logger'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/sprockets-rails-3.2.1/lib/sprockets/rails/task.rb:67:in `block (2 levels) in define'
       /tmp/build_d8e573ed04088d976e972a9313d20c49/vendor/bundle/ruby/2.6.0/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
       Tasks: TOP => assets:precompile
       (See full trace by running task with --trace)
```

**Comment by alvesjtiago** [comment]
Thanks for reporting this @bf4. Since `rails-ujs` is available from rails 5.1 onwards and AFAIK there's no way to conditionally add it to the dependencies, I will just mention it on the README for now. Hopefully in the future I'll be able to rewrite it and not depend on `rails-ujs`.

**Comment by bf4** [comment]
good enough :)
